src/librawspeed/tiff/TiffEntry.cpp

```diff
@@ -189,12 +189,13 @@ float TiffEntry::getFloat(uint32 num) const {
   }
 }
 
-string TiffEntry::getString() const {
+string TiffEntry::getString(bool trimmed) const {
   if (type != TIFF_ASCII && type != TIFF_BYTE)
     ThrowTPE("TIFF, getString: Wrong type 0x%x encountered. Expected Ascii or Byte", type);
 
   const char* s = data.peekString();
-  return string(s, strnlen(s, count));
+  string result = string(s, strnlen(s, count));
+  return trimmed ? trimSpaces(result) : result;
```

Comment:
dangerous how? rstest runs through and this change possibly detected a bug with the two sonys.